packages/next-app/src/app/stableswapsimulator/(components)/PoolParametersForm.tsx

```diff
@@ -1,13 +1,87 @@
 "use client";
 
-import { useStableSwap } from "#/contexts/StableSwapContext";
+import { capitalize } from "@balancer-pool-metadata/shared";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+import Button from "#/components/Button";
+import { Input } from "#/components/Input";
+import { AnalysisData, useStableSwap } from "#/contexts/StableSwapContext";
 
 export default function PoolParametersForm() {
-  const { initialData } = useStableSwap();
+  const { initialData, setInitialData, newPoolImportedFlag } = useStableSwap();
+  const {
+    register,
+    handleSubmit,
+    clearErrors,
+    setValue,
+    formState: { errors },
+  } = useForm();
+
+  const inputParameters = (field: keyof AnalysisData) => {
+    const label = field.replace(/([a-z])([A-Z])/g, "$1 $2").toLowerCase();
+    if (field == "tokens") return; // TODO: BAL 386
+    return {
+      ...register(field, { required: true, value: initialData?.[field] }),
+      label: capitalize(label),
+      type: "number",
+      placeholder: `Define the initial ${label}`,
+      value: initialData?.[field],
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value =
+          e.target.value == "" ? undefined : parseFloat(e.target.value);
+        if (value == undefined) {
+          setInitialData({
+            ...initialData,
+            [field]: undefined,
+          });
+          return;
+        }
+        if (value < 0) return;
+        setInitialData({
+          ...initialData,
+          [field]: value,
+        });
+      },
+    };
+  };
```

Comment:
curious to learn more about what value you had in mind to add with this abstraction -- I think this is more of a style preference, but I really dislike this pattern. Its abstraction adds indirection and is coupled with the implementation of your fields -- you're predefining  a placeholder pattern and a number-typed field, as well as with a state setter (`setInitialData`) and spreading the `register` from `react-hook-form` within it which I think could be bug prone since I don't think we could guarantee this function will be defined every time the component rerenders, and I don't know what are the expectations of react-hook-form about when it's gonna be called / by whom.

Reply:
I'll open a new issue to solve this. I checked the way that you implemented but when you import a pool is not change the fields.